metadata/register.py

```diff
@@ -565,16 +710,43 @@ def main():
         else:
             if "bioformats2raw.layout" in zattrs and zattrs["bioformats2raw.layout"] == 3:
                 print("Registering: bioformats2raw.layout")
-                series = 0
-                series_exists = True
-                while series_exists:
-                    try:
-                        print("Checking for series:", series)
-                        obj = register_image(conn, store, args, None, image_path=str(series))
-                        objs.append(obj)
-                    except FileNotFoundError:
-                        series_exists = False
-                    series += 1
+                zarr_name = args.uri.rstrip("/").split("/")[-1]
+                if args.name:
+                    zarr_name = args.name
+                # try to load OME/METADATA.ome.xml
+                omexml_bytes = get_omexml_bytes(store)
+                if omexml_bytes is not None:
+                    print("Importing OME/METADATA.ome.xml")
+                    rsp = full_import(conn.c, omexml_bytes, args.wait)
+                    for series, p in enumerate(rsp.pixels):
+                        # set external info. NB: order of pixels MUST match the series 0, 1, 2...
+                        image = conn.getObject("Image", p.image.id.val)
+                        image_path = str(series)
+                        image_attrs = load_attrs(store, image_path)
+                        # pixels_type is only used if we have *incomplete* `omero` metadata
+                        sizes, pixels_type = parse_image_metadata(store, image_attrs, image_path)
+                        rnd_def = set_rendering_settings(conn, image, image_attrs, pixels_type)
```

Comment:
This doesn't work, because set_rendering_settings looks in the zattr not in the METADATA.ome.xml. Should probably not be necessary anyway. Hopfully importing the METADATA.ome.xml should also create the rendering settings, if specified in the xml!?